src/stores/SetupEncryptionStore.js

```diff
@@ -119,6 +131,14 @@ export class SetupEncryptionStore extends EventEmitter {
             this.verificationRequest.off("change", this.onVerificationRequestChange);
             this.verificationRequest = null;
             this.emit("update");
+        } else if (this.verificationRequest.phase === VERIF_PHASE_DONE) {
+            this.verificationRequest.off("change", this.onVerificationRequestChange);
+            this.verificationRequest = null;
+            // At this point, the verification has finished, we just need to wait for
```

Comment:
Is there any way these could happen in the opposite order? As in, user trust / cross-signing ready somehow happens before the verification is done? (If it did happen, it looks like we'd go to done and then back to busy forever.)

Reply:
Hmm - I wouldn't have thought so, although it's probably easiest to just check rather than reason through and rely on them always happening in the right order.

Reply:
Submitted as https://github.com/matrix-org/matrix-react-sdk/pull/4539 to develop and added to this PR